contribs/gnostats/agent/agent.go

```diff
@@ -0,0 +1,81 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gnolang/gnostats/proto"
+)
+
+// config holds both the injected Gno node RPC client and the Hub gRPC client.
+type config struct {
```

Comment:
Nitpick: maybe these should be args for the agent constructor, and contained within the agent, and not the configuration. We can make the poll interval have a default value, but make it configurable through an option. The hub client proto and rpc client should be required

Reply:
Okay, indeed you are right that's better. I moved the interval into an option using the `WithOption` pattern if that's what you meant.